internal/controller/apiextensions/composite/composition_transforms.go

```diff
@@ -485,3 +478,12 @@ var conversions = map[conversionPair]func(any) (any, error){
 		return i.(float64) == float64(1), nil
 	},
 }
+
+// typeOf is just a wrapper around reflect.TypeOf, but will return a string
+// representation of nil if the input is nil instead of panicking.
+func typeOf(input any) string {
+	if input == nil {
+		return fmt.Sprintf("%T", input)
```

Comment:
I played a bit with `fmt.Sprintf("%T, v)` and it seems that it works fine for non-nil inputs as well... could we use it then for all values:

```
func typeOf(input any) string {
  return fmt.Sprintf("%T", input)
}
```

Also, could we place this function in some util package and use it everywhere needed?

Reply:
If we just use fmt.Sprintf i wouldn't bother adding that to an utils package. Will do.

Reply:
removed and refactored a bit, can you have a look @pedjak ?